documentation/src/main/asciidoc/user_guide/query.adoc

```diff
@@ -251,77 +759,191 @@ Assert.assertEquals(cq.getResultSize(), 1);
+SearchManager searchManager = Search.getSearchManager(cache);
+Query query = searchManager.buildQueryBuilderForClass(Book.class).get()
+            .keyword().wildcard().onField("description").matching("*test*").createQuery();
+CacheQuery<Book> cacheQuery = searchManager.getQuery(query);
+----
 
-For full documentation on clustering the Lucene engine, refer to the link:http://docs.jboss.org/hibernate/stable/search/reference/en-US/html_single/#search-configuration[Hibernate Search documentation] to properly configure it clustered.
 
-=== Clustering the Index in Infinispan
-Again the configuration details are in the Hibernate Search reference, in particular in the link:http://docs.jboss.org/hibernate/stable/search/reference/en-US/html_single/#infinispan-directories[infinispan-directories] section. This backend will by default start a secondary Infinispan CacheManager, and optionally take another Infinispan configuration file: don't reuse the same configuration or you will start grids recursively! It is currently not possible to share the same CacheManager.
+====== Using the Hibernate Search DSL
 
-=== Rebuilding the Index
-Occasionally you might need to rebuild the Lucene index by reconstructing it from the data stored in the Cache. You need to rebuild the index if you change the definition of what is indexed on your types, or if you change for example some _Analyzer_ parameter, as Analyzers affect how the index is defined. Also, you might need to rebuild the index if you had it destroyed by some system administration mistake. To rebuild the index just get a reference to the MassIndexer and start it; beware if might take some time as it needs to reprocess all data in the grid!
+The Hibernate Search DSL can be used to create the Lucene Query, example:
 
 [source,java]
 ----
+import org.infinispan.query.Search;
+import org.infinispan.query.SearchManager;
+import org.apache.lucene.search.Query;
+
+Cache<String, Book> cache = ...
+
 SearchManager searchManager = Search.getSearchManager(cache);
-searchManager.getMassIndexer().start();
+
+Query luceneQuery = searchManager
+                         .buildQueryBuilderForClass(Book.class).get()
+                         .range().onField("year").from(2005).to(2010)
+                         .createQuery();
+
+List<Object> results = searchManager.getQuery(luceneQuery).list();
 ----
 
-TIP: This is also available as a `start` JMX operation on the link:{javadocroot}/jmxComponents.html#MassIndexer[MassIndexer MBean]
-registered under the name `org.infinispan:type=Query,manager="{name-of-cache-manager}",cache="{name-of-cache}",component=MassIndexer`.
+For a detailed description of the query capabilities of this DSL, see the relevant section of the link:https://docs.jboss.org/hibernate/stable/search/reference/en-US/html_single/#section-building-lucene-queries[Hibernate Search manual].
```

Comment:
Will sort it out later once variables are available for both lucene and hs https://issues.jboss.org/browse/ISPN-7560